File: app/src/main/java/org/nitri/opentopo/nearby/api/mediawiki/MediaWikiApi.java
package org.nitri.opentopo.nearby.api.mediawiki;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MediaWikiApi {

    @GET("w/api.php")
    Call<MediaWikiResponse> getNearbyPages(@Query("action") String action,
                                           @Query("prop") String prop,
                                           @Query("colimit") int colimit,
                                           @Query("piprop") String piprop,
                                           @Query("pithumbsize") int pithumbsize,
                                           @Query("pilimit") int pilimit,
                                           @Query("wbptterms") String wbptterms,
                                           @Query("generator") String generator,
                                           @Query("ggscoord") String ggscoord,
                                           @Query("ggsradius") int ggsradius,
                                           @Query("ggslimit") int ggslimit,
                                           @Query("inprop") String inprop,
                                           @Query("format") String format);
}

File: app/src/main/java/org/nitri/opentopo/nearby/api/mediawiki/MediaWikiApi.kt
package org.nitri.opentopo.nearby.api.mediawiki

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface MediaWikiApi {
    @GET("w/api.php")
    fun getNearbyPages(
        @Query("action") action: String?,
        @Query("prop") prop: String?,
        @Query("colimit") colimit: Int,
        @Query("piprop") piprop: String?,
        @Query("pithumbsize") pithumbsize: Int,
        @Query("pilimit") pilimit: Int,
        @Query("wbptterms") wbptterms: String?,
        @Query("generator") generator: String?,
        @Query("ggscoord") ggscoord: String?,
        @Query("ggsradius") ggsradius: Int,
        @Query("ggslimit") ggslimit: Int,
        @Query("inprop") inprop: String?,
        @Query("format") format: String?
    ): Call<MediaWikiResponse?>?
}
